flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/codegen/calls/ScalarFunctionCallGen.scala

```diff
@@ -66,7 +67,18 @@ class ScalarFunctionCallGen(scalarFunction: ScalarFunction) extends CallGenerato
       boxedTypeTermForType(returnType)
     }
     val resultTerm = ctx.addReusableLocalVariable(resultTypeTerm, "result")
-    val evalResult = s"$functionReference.eval(${parameters.map(_.resultTerm).mkString(", ")})"
+    val evalResult =
+      if (returnType.getTypeRoot == LogicalTypeRoot.TIMESTAMP_WITH_LOCAL_TIME_ZONE
+          && (resultClass == classOf[Long] || resultClass == classOf[JLong])) {
+        // Convert Long to SqlTimestamp if the UDX's returnType is
```

Comment:
do we have to treat this as special case here?

Reply:
will add a Long <-> Instant converter to support this.